### PR TITLE
refactor: improve variable logic

### PR DESCRIPTION
**Pull request by retrogtx**
## What does this PR do?

- Refactors the forEachAppDir function in packages/app-store-cli/src/build.ts to use async/await and dynamic imports
- Improves error handling for file imports in the app store CLI build process

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox. NA.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

### CODE REVIEW

**Comment by Praashh** [comment]
Hey @retrogtx, there are some type issues in the ``` app-store``` package, can you please check and fix those?

**Comment by retrogtx** [comment]
> Hey @retrogtx, there are some type issues in the ` app-store` package, can you please check and fix those?

hi! by type issues do you mean the line `import prettier from "prettier";` ?

**Comment by Praashh** [comment]
> > Hey @retrogtx, there are some type issues in the ` app-store` package, can you please check and fix those?
> 
> hi! by type issues do you mean the line `import prettier from "prettier";` ?

no, I am seeing some test cases are failing, for eg: in ```app-store/BookingPageTagManager.tsx:106:17 ``` eventTypeAppData' is possibly undefined. so please try to add some checks and types.

**Comment by retrogtx** [comment]
``

> there are some type issues in the ` app-store` package, can you please check and fix those?

yep just got it rn

**Review by Praashh — APPROVED** [review]

**Comment by retrogtx** [comment]
seems like it works now, I was looking in the wrong place for nearly twenty minutes
adding back `//@ts-ignore` made it work. I think it'll be needed as long as the prettier error isn't fixed
all good now

**Comment by Praashh** [comment]
> seems like it works now, I was looking in the wrong place for nearly twenty minutes adding back `//@ts-ignore` made it work. I think it'll be needed as long as the prettier error isn't fixed all good now

I'll give it a final review, seems good to me.

**Review by Praashh — DISMISSED** [review]
looks good to me.

**Comment by AliPythonDev** [comment]
Hey @retrogtx 
A couple of suggestions:
- Ensure that using arrow functions in place of the traditional function syntax is compatible with any other parts of the codebase where context (this) might be important. In this case, it seems fine since there’s no explicit need for the this binding.
- The change to const isInWatchMode = process.argv[2] === "--watch"; is neat and removes the extra if block, making it cleaner.

Overall, great job improving readability and consistency!

File: packages/app-store-cli/src/build.ts [hunk 1]
```diff
@@ -13,25 +13,17 @@ import type { AppMeta } from "@calcom/types/App";
...
     ...prettierConfig,
   });
 
-const getVariableName = function (appName: string) {
-  return appName.replace(/[-.]/g, "_");
-};
+const getVariableName = (appName: string) => appName.replace(/[-.]/g, "_");
 
-const getAppId = function (app: { name: string }) {
-  // Handle stripe separately as it's an old app with different dirName than slug/appId
```
Comment (keithwillcode) [inline packages/app-store-cli/src/build.ts:32]: While comments can be stale and/or not that useful, this one actually is. We should keep it in. Let's just add an 
// INFO: Handle stripe separately as it's an old app with different dirName than slug/appId above the function definition.

**Review by keithwillcode — CHANGES_REQUESTED** [review]

File: packages/app-store-cli/src/build.ts (hunk 1 above)
Comment (retrogtx) [reply packages/app-store-cli/src/build.ts:32]: sounds right, just did that

**Comment by retrogtx** [comment]
> Hey @retrogtx A couple of suggestions:
> 
> * Ensure that using arrow functions in place of the traditional function syntax is compatible with any other parts of the codebase where context (this) might be important. In this case, it seems fine since there’s no explicit need for the this binding.
> * The change to const isInWatchMode = process.argv[2] === "--watch"; is neat and removes the extra if block, making it cleaner.
> 
> Overall, great job improving readability and consistency!

oh refactoring was not the main goal, I was trying to fix the prettier import error

**Review by keithwillcode — APPROVED** [review]

**Pull request merged**